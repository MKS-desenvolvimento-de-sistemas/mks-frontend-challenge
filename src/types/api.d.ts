export interface IProduct {
  id: number;
  name: string;
  brand: string;
  description: string;
  price: string;
  photo: string;
  createdAt: string;
  updatedAt: string;
}

export interface IProductCart extends IProduct {
  quantity: number;
}

export interface IProducts {
  products: IProduct[];
}
