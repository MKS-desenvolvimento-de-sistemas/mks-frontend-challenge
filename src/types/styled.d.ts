import {} from "styled-components";
import { theme } from "../styles/themes/";

declare module "styled-components" {
  export interface DefaultTheme extends theme {}
}
