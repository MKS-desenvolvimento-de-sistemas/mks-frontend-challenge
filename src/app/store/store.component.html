<div class="store">
    <h1>Store</h1>
    <div *ngIf="isLoading">Loading...</div>
    <app-product-card *ngFor="let product of products" [product]="product" (addToCartEvent)="addToCart($event)"></app-product-card>
  </div>
  