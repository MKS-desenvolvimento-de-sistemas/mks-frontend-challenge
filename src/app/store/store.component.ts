import { Component, OnInit } from '@angular/core';
import { ApiService, Product } from '..//services/api.service';

@Component({
  selector: 'app-store',
  templateUrl: './store.component.html',
  styleUrls: ['./store.component.scss']
})
export class StoreComponent implements OnInit {
  products: Product[] = [];
  isLoading = true;

  constructor(private apiService: ApiService) {}

  ngOnInit() {
    this.apiService.fetchProducts().then(products => {
      this.products = products;
      this.isLoading = false;
    });
  }

  addToCart(product: Product) {
    // Implementar a l√≥gica de adicionar ao carrinho
    console.log('Adicionar ao carrinho:', product);
  }
}
