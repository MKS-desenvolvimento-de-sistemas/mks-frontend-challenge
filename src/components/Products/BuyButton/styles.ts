import styled from "styled-components";

export const Button = styled.button`
  width: 100%;
  padding: 8.06px 0 12.55px 0;
  background: var(--terciary-color);
  border-radius: 0px 0px 8px 8px;

  display: flex;
  justify-content: center;
  align-items: center;
  gap: 16.12px;

  font-weight: 600;
  font-size: 14px;
  line-height: 18px;
  text-transform: uppercase;
  color: var(--primary-color-100);

  &:hover {
    filter: opacity(0.8);
  }

  @media (min-width: 760px) {
    gap: 14px;
    padding: 7px 0 10.91px 0;
  }
`;
