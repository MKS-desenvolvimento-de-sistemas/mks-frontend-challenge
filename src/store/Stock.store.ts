import {createSlice} from '@reduxjs/toolkit';

const carrinho = createSlice({
	name: 'carrinho',
	initialState: {
		counter: 0,
		cartOpen: 'none',
	},
	reducers: {
		increment(state) {
			state.counter += 1;
		},
		decrement(state) {
			state.counter -= 1;
		},
		abrirCarrinho(state) {
			state.cartOpen = 'flex';
		},
		fecharCarrinho(state) {
			state.cartOpen = 'none';
		},
	},
});

export const {
	decrement,
	increment,
	abrirCarrinho,
	fecharCarrinho,
} = carrinho.actions;
export default carrinho.reducer;
