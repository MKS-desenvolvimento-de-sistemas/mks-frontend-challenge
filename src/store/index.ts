import { createSlice } from "@reduxjs/toolkit";
import { configureStore } from "@reduxjs/toolkit";
import { ICartItem } from "../pages/Cart";
const initialState = { items: [] } as any;

const cartSlice = createSlice({
  name: "carrinho",
  initialState,
  reducers: {
    addToCart(state, action) {
      state.items.push(action.payload);
    },

    addAdditional(state, action) {
      if (!state.items.some((item) => item.id === action.payload.id)) return;
      const item = state.items.find((item) => item.id === action.payload.id);

      state.items.push(item);
    },
    decreaseItem(state, action) {
      const remainingItems = state.items.filter(
        (item) => item.id === action.payload.id
      );

      if (remainingItems.length === 1) return;

      const itemIndex = state.items.findIndex(
        (el) => el.id === action.payload.id
      );

      state.items.splice(itemIndex, 1);
    },
    deleteItem(state, action) {
      const arr = state.items.filter((item) => item.id !== action.payload.id);
      state.items = arr;
      // do {
      //   const itemIndex = state.items.findIndex(
      //     (el) => el.id === action.payload.id
      //   );

      //   state.items.splice(itemIndex, 1);
      // } while (state.items.some((item) => item.id === action.payload.id));
    },
    deleteAllItems(state) {
      state.items = [];
    },
  },
});

const store = configureStore({
  reducer: cartSlice.reducer,
});

export const cartActions = cartSlice.actions;

export default store;
