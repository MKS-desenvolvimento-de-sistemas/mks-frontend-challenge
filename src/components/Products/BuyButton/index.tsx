import * as Styles from "./styles";
import { FiShoppingBag } from "react-icons/fi";
import { IProduct } from "../../../types/api";

// REDUX IMPORT'S
import { add } from "../../../redux/Cart";
import { useDispatch } from "react-redux";

interface IBuyButtonProps {
  product: IProduct;
}

const BuyButton = ({ product }: IBuyButtonProps) => {
  const dispatch = useDispatch();

  return (
    <Styles.Button
      type="button"
      title="Adicionar ao Carrinho"
      onClick={() => dispatch(add(product))}
    >
      <FiShoppingBag size="16px" />
      Comprar
    </Styles.Button>
  );
};

export default BuyButton;
