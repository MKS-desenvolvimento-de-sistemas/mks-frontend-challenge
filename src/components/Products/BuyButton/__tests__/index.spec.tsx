import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import BuyButton from "../";
import renderer from "react-test-renderer";
import { IProduct } from "../../../../types/api";
import { Provider } from "react-redux";
import store from "../../../../redux/";

const productExample: IProduct = {
  id: 1,
  name: "EXEMPLO 1",
  brand: "Exemplo 1",
  description: "Exemplo 1",
  price: "200",
  photo: "/vercel.svg",
  updatedAt: "2022",
  createdAt: "2022",
};

describe("BuyButton Component", () => {
  it("should render the BuyButton correctly", () => {
    render(
      <Provider store={store}>
        <BuyButton product={productExample} />
      </Provider>
    );

    const button = screen.getByTestId("buy-button");

    expect(button).toHaveTextContent("Comprar");
    expect(button.getAttribute("type")).toEqual("button");
    expect(button.getAttribute("title")).toEqual("Adicionar ao Carrinho");
  });

  it("should match the snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BuyButton product={productExample} />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
