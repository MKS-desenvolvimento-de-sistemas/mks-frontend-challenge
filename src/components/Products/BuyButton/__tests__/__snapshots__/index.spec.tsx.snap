// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyButton Component should match the snapshot 1`] = `
<button
  className="sc-bczRLJ clkVIK"
  data-testid="buy-button"
  onClick={[Function]}
  title="Adicionar ao Carrinho"
  type="button"
>
  <svg
    fill="none"
    height="16px"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="16px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
    />
    <line
      x1="3"
      x2="21"
      y1="6"
      y2="6"
    />
    <path
      d="M16 10a4 4 0 0 1-8 0"
    />
  </svg>
  Comprar
</button>
`;
